### PR TITLE
chore(MenuButton): Convert to RFC

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

# BREAKING CHANGES

Converting `MenuButton` from class component to functional. Any props are not longer passed to styles function.

Related to #12237

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by assuncaocharles** [comment]
@layershifter @pompomon Tests are broken because attributes behaviors are not being applied to `trigger`. Still trying to find, but any of you has any idea?

**Comment by assuncaocharles** [comment]
Everything ok on this one @layershifter @pompomon

**Comment by assuncaocharles** [comment]
Gentle reminder @layershifter @pompomon

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 1]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
-    contextMenu: PropTypes.bool,
-  };
+    rtl: context.rtl,
+  });
 
-  static defaultProps: MenuButtonProps = {
-    accessibility: menuButtonBehavior,
-    align: 'start',
-    position: 'below',
+  const popupProps = {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:185]:
```suggestion
  const popupProps: PopupProps = {
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 2]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
+    offset,
+    on,
+    onOpenChange,
+    styles,
+    overflowBoundary,
+    pointing,
+    position,
+    positionFixed,
+    tabbableTrigger,
+    styles: stylesProp,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:142]:
```suggestion
    styles: props.styles,
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 3]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
+    offset,
+    on,
+    onOpenChange,
+    open,
+    overflowBoundary,
+    pointing,
+    position,
+    positionFixed,
+    tabbableTrigger,
+    styles: stylesProp,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:202]:
```suggestion
    styles: props.styles,
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 4]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
-    };
-  }
-
-  triggerRef = React.createRef<HTMLElement>();
-  menuRef = React.createRef<HTMLElement>();
+  const { classes, styles: resolvedStyles } = useStyles<MenuButtonStylesProps>(MenuButton.displayName, {
+    className: menuButtonClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      styles,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:213]:
```suggestion
      styles: props.styles,
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 5]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
-    _.invoke(this.props, 'onOpenChange', e, { ...this.props, open });
-    this.setState({ open }, callback);
+  const handleOpenChange = (e: React.SyntheticEvent, open: boolean, callback?: () => void) => {
+    _.invoke(props, 'onOpenChange', e, { ...props, open });
+    setOpen(open);
+    callback && callback();
   };
 
-  handleMenuOverrides = (predefinedProps?: MenuProps) => ({
+  const handleMenuOverrides = (predefinedProps?: MenuProps) => ({
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:234]:
```suggestion
  const handleMenuOverrides = (predefinedProps: MenuProps) => ({
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 6]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
+  );
+  setEnd();
+  return element;
+};
+
+MenuButton.displayName = 'MenuButton';
+
+MenuButton.propTypes = {
+  ...commonPropTypes.createCommon({
+    as: true,
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:302]:
```suggestion
```

File: packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx [hunk 7]
```diff
@@ -98,218 +107,250 @@ export const menuButtonSlotClassNames: MenuButtonSlotClassNames = {
...
+
+MenuButton.defaultProps = {
+  accessibility: menuButtonBehavior,
+  align: 'start',
+  position: 'below',
+};
+
+MenuButton.handledProps = Object.keys(MenuButton.propTypes) as any;
+
+// MenuButton.autoControlledProps = ['open'];
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/MenuButton/MenuButton.tsx:352]:
```suggestion
```

**Pull request merged**